File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:

        output =[]
        n = len(nums)
        if n<=k:
            return [max(nums)]
        currmax = max(nums[0:k])
        output.append(currmax)
        L,R=0,k-1
        while(R+1<n):
            R+=1
            if nums[R]>currmax:
                currmax = nums[R]
                L=R
            if R-L+1 > k:
                L+=1
                currmax = max(nums[L:R+1])
            output.append(currmax)
        return output

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int n = nums.size();
        int L = 0, R = n-1;
        int mid = (L+R)/2;
        int res;
        if(nums[0]<=nums[R]) return nums[0];
        while(L<R){
            cout<<L<<" "<<mid<<" "<< R<<endl;
            if(nums[L]>nums[mid]) R=mid;
            else if (nums[R]< nums[mid]) L = mid+1;
            else {
                return nums[L];
            }
            mid = (L+R)/2;
        }
                    cout<<L<<" "<<mid<<" "<< R<<endl;
        return nums[mid];
    }
};
